apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -1042,15 +1029,19 @@ func (api *ProvisionerAPI) prepareOrGetContainerInterfaceInfo(
 // prepareContainerAccessEnvironment retrieves the environment, host machine, and access
 // for working with containers.
 func (api *ProvisionerAPI) prepareContainerAccessEnvironment(ctx context.Context) (environs.Environ, *state.Machine, common.AuthFunc, error) {
-	env, err := environs.GetEnviron(ctx, api.configGetter, environs.New)
+	modelInfo, err := api.modelInfoService.GetModelInfo(ctx)
 	if err != nil {
-		return nil, nil, nil, errors.Trace(err)
+		return nil, nil, nil, fmt.Errorf("getting model info: %w", err)
 	}
-	// TODO(jam): 2017-02-01 NetworkingEnvironFromModelConfig used to do this, but it doesn't feel good
-	if env.Config().Type() == "dummy" {
+	if modelInfo.CloudType == "dummy" {
```

Comment:
todo: remove this, the environ won't be supported when we ask for it.

Reply:
done